data/lescure2020clinical/lescure2020clinical-extraction.md

```diff
@@ -0,0 +1,85 @@
+
+First, we `import` python modules needed:
+
+```python
+#import modules;
+import yaml
+import pandas as pd
+
+#functions to add folded blocks and literal blocks;
+class folded_str(str): pass
+class literal_str(str): pass
+
+def folded_str_representer(dumper, data):
+    return dumper.represent_scalar(u'tag:yaml.org,2002:str', data, style='>')
+def literal_str_representer(dumper, data):
+    return dumper.represent_scalar(u'tag:yaml.org,2002:str', data, style='|')
+
+yaml.add_representer(folded_str, folded_str_representer)
+yaml.add_representer(literal_str, literal_str_representer)
+```
+
+Raw data ([Viral_Loads](https://github.com/shedding-hub/shedding-hub/blob/main/data/lescure2020clinical/Viral_Loads.csv)), which is stored on [Shedding Hub](https://github.com/shedding-hub), will be loaded and cleaned to match the most updated [schema](https://github.com/shedding-hub/shedding-hub/blob/main/data/.schema.yaml).
+
+```python
+#load the data;
+Lescure2020 = pd.read_csv("Viral_Loads.csv")
+#Subset the data for Lescure et al. 2020;
+Lescure2020 = Lescure2020[Lescure2020["cov_study"]==4]
+#correct some doa errors;
+Lescure2020.loc[(Lescure2020["ID"]=="E3") & (Lescure2020["dao"]==13),"dao"]=14
+Lescure2020.loc[(Lescure2020["ID"]=="E3") & (Lescure2020["dao"]==15),"dao"]=16
+#some data cleaning to match the schema;
+Lescure2020.loc[Lescure2020["cens"]==0,"VL"]=10**Lescure2020.loc[Lescure2020["cens"]==0,"VL"]
+Lescure2020.loc[Lescure2020["cens"]==1,"VL"]="negative"
+#The positive but not quantifiable data are from Figure 2 in the paper;
+Lescure2020.loc[(Lescure2020["ID"]=="E3") & (Lescure2020["dao"].isin([14,16])),"VL"]="positive"
+Lescure2020.loc[(Lescure2020["ID"]=="E5") & (Lescure2020["dao"]==8),"VL"]="positive"
+
+#add data from the patient 2;
+P2 = {
+    "ID": ['E2','E2','E2','E2','E2'],
+    "dao": [0,2,5,11,13],
+    "VL": ['positive','negative','negative','negative','negative']
+}
+dat_P2 = pd.DataFrame(P2)
+
+Lescure2020 = pd.concat([Lescure2020,dat_P2])
+```
+
+Finally, the data is formatted and output as a YAML file.
+
+```python
+#Patient information from Table 1 and Figure 2 in Lescure et al. (2020)
+patients_info = pd.DataFrame(dict(ID=['E1','E2','E3', 'E4', 'E5'],
+                                  age=[31, 48, 80, 30, 46],
+                                  sex=['male','male','male','female','female'],
+                                  day0=[6,9,8,2,2]))
+
+participant_list = [dict(attributes=dict(age=patients_info.loc[patients_info["ID"]==i,"age"].item(),
+                                         sex=patients_info.loc[patients_info["ID"]==i,"sex"].item(),),
+                         measurements=[dict(analyte="naso_swab_SARSCoV2",
+                                             time=int(Lescure2020.loc[j,"dao"].item()+patients_info.loc[patients_info["ID"]==i,"day0"].item()),
```

Comment:
Could be useful to have a small comment here to explain what the `dao` and `day0` variables mean.

Reply:
Added a comment: `#the variable "dao" is "the day after first positive" used in Goyal et al. (2020) and "day0" is "the sampling day after onset". "dao" + "day0" will be "day after symptom onset".`

Reply:
Great, thank you!